Application/components/FriendsColumn/RemoveFriendModal.tsx

```diff
@@ -0,0 +1,148 @@
+import React, { useState } from 'react';
+import { Modal, Text, Stack, Button, useMantineTheme, MultiSelect, Tooltip, Paper } from '@mantine/core';
+import { IconUserMinus } from '@tabler/icons-react';
+import { useUser } from '@clerk/nextjs';
+import { useCache } from '@/contexts/queryCacheContext';
+import { useFriendList } from '@/hooks/useFriendList';
+import { showNotification } from '@mantine/notifications';
+
+interface Friend {
+  id: string;
+  username: string;
+}
+
+interface RemoveFriendResponse {
+  message: string;
+}
+
+export function RemoveFriendModal(): JSX.Element {
+  const { user } = useUser();
+  const theme = useMantineTheme();
+  const [opened, setOpened] = useState<boolean>(false);
+  const { lastFetched, setLastFetched } = useCache();
+  const [selectedFriend, setSelectedFriend] = useState<string[]>([]);
+  const [errorMessages, setErrorMessages] = useState<{ friends: string }>({ friends: '' });
+  const { list: friendsList } = useFriendList({ lastFetched, setLastFetched });
+
+  const friendOptions = friendsList.map((friend: Friend) => ({
+    value: friend.id,
+    label: friend.username,
+  }));
+
+  const handleOpenModalClick = () => {
+    if (friendsList.length > 0) {
+      setOpened(true);
+    } else {
+      showNotification({
+        title: 'No Friends to Remove',
+        message: 'You currently have no friends to remove.😢',
+        color: 'red',
+      });
+    }
+  };
+
+  const handleRemoveFriendClick = async (): Promise<void> => {
+    if (selectedFriend.length === 0) {
```

Comment:
Great client-side validation! This is super important to enhancing the user experience and I'm happy to see you included that!